File: old_WeatherAlerts/ClientApp/src/app/weather-api/models/alert-properties.ts
import { IAlertProperties } from '../interfaces/ialert-properties';

export class AlertProperties implements IAlertProperties {

    id: string;
    status: string;
    category: string;
    severity: string;
    urgency: string;
    event: string;
    senderName: string;
    headline: string;
    description: string;
    response: string;
    messageType: string;
    certainty: string;
    areaDesc: string;
    sent: Date;
    effective: Date;
    onset: Date;
    expires: Date;
    ends: Date;

    constructor() {
        this.id = '';
        this.status = '';
        this.severity = '';
        this.urgency = '';
        this.event = '';
        this.senderName = '';
        this.headline = '';
        this.description = '';
        this.response = '';
        this.messageType = '';
        this.certainty = '';
        this.areaDesc = '';
        this.sent = new Date();
        this.effective = new Date();
        this.onset = new Date();
        this.expires = new Date();
        this.ends = new Date();
    }
}

File: WeatherAlertsApi/WeatherAlertsApi.Core/ApiModels/WeatherApi/AlertProperties.cs
using System.Text.Json.Serialization;

namespace WeatherAlertsApi.Core.ApiModels.WeatherApi;

public class AlertProperties
{
    public string? id { get; set; }
    public string? status { get; set; }
    public string? category { get; set; }
    public string? severity { get; set; }
    public string? urgency { get; set; }
    public string? Event { get; set; }
    public string? senderName { get; set; }
    public string? headline { get; set; }
    public string? description { get; set; }
    public string? areaDesc { get; set; }
    public DateTime sent { get; set; }
    public DateTime effective { get; set; }
    public DateTime onset { get; set; }
    public DateTime expires { get; set; }
    public DateTime? ends { get; set; }

}